### PR TITLE
Implements API v2

### LINKED ISSUE #23: Read configurations from different files
See https://github.com/benchflow/drivers-maker/issues/20

### LINKED ISSUE #32: Remove the API v1

### PR DESCRIPTION
**Pull request by simonedavico**
- Implements an improved API to deploy, run and poll experiments for their status.
- Resolves #32, removing API v1;
- Resolves #23;
- Removes MemSQL usage from test Dockerfile, and adds MYSQL

<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/benchflow/experiments-manager/pull/38%23issuecomment-230039933%22%2C%20%22https%3A//github.com/benchflow/experiments-manager/pull/38%23issuecomment-230357041%22%2C%20%22https%3A//github.com/benchflow/experiments-manager/pull/38%23discussion_r68979621%22%2C%20%22https%3A//github.com/benchflow/experiments-manager/pull/38%23discussion_r68979852%22%2C%20%22https%3A//github.com/benchflow/experiments-manager/pull/38%23discussion_r68983515%22%2C%20%22https%3A//github.com/benchflow/experiments-manager/pull/38%23discussion_r69492820%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/experiments-manager/pull/38%23issuecomment-230039933%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40simonedavico%20ping%20me%20when%20done%22%2C%20%22created_at%22%3A%20%222016-07-01T20%3A26%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-04T21%3A42%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%203b21bace0282cb21b869cdab245887b0b4d56b08%20application/src/main/java/cloud/benchflow/experimentsmanager/resources/lifecycle/v2/ExperimentStatusResource.java%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/experiments-manager/pull/38%23discussion_r68979621%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Remove%20the%20%60v2%60%20as%20well%20keeping%20only%20the%20actual%20resources.%22%2C%20%22created_at%22%3A%20%222016-06-29T16%3A23%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20don%27t%20know.%20We%20will%20probably%20need%20a%20%60v3%60%20for%20the%20new%20interactions%20from%20the%20orchestrator%20and%20the%20removal%20of%20the%20%60deploy%60%20phase.%22%2C%20%22created_at%22%3A%20%222016-06-29T16%3A25%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4446853%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/simonedavico%22%7D%7D%2C%20%7B%22body%22%3A%20%22We%20can%20also%20do%20this%20change%20in%20a%20different%20branch%2C%20and%20when%20ready%20we%20merge%20everything%20back%20to%20the%20devel%20branch%2C%20removing/updating%20the%20current%20API.%22%2C%20%22created_at%22%3A%20%222016-06-29T16%3A46%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-04T21%3A42%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20application/src/main/java/cloud/benchflow/experimentsmanager/resources/lifecycle/v2/ExperimentStatusResource.java%3AL1-8%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 3b21bace0282cb21b869cdab245887b0b4d56b08 application/src/main/java/cloud/benchflow/experimentsmanager/resources/lifecycle/v2/ExperimentStatusResource.java 1'></a> &nbsp;<b><a href='https://github.com/benchflow/experiments-manager/pull/38#discussion_r68979621'>File: application/src/main/java/cloud/benchflow/experimentsmanager/resources/lifecycle/v2/ExperimentStatusResource.java:L1-8</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Remove the `v2` as well keeping only the actual resources.
- <a href='https://github.com/simonedavico'></a> I don't know. We will probably need a `v3` for the new interactions from the orchestrator and the removal of the `deploy` phase.
- <a href='https://github.com/VincenzoFerme'></a> We can also do this change in a different branch, and when ready we merge everything back to the devel branch, removing/updating the current API.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/benchflow/experiments-manager/pull/38#issuecomment-230039933'>General Comment</a></b>
- <a href='https://github.com/VincenzoFerme'></a> @simonedavico ping me when done

<a href='https://www.codereviewhub.com/benchflow/experiments-manager/pull/38?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/benchflow/experiments-manager/pull/38?approve=1'></a>&nbsp;<a href='https://github.com/benchflow/experiments-manager/pull/38'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by VincenzoFerme** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/benchflow/experiments-manager/pull/38'>Refresh</a>.<a href='#crh-update-none'></a>

File: application/src/main/java/cloud/benchflow/experimentsmanager/resources/lifecycle/v2/ExperimentStatusResource.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package cloud.benchflow.experimentsmanager.resources.lifecycle.v2;
```
Comment (VincenzoFerme) [inline application/src/main/java/cloud/benchflow/experimentsmanager/resources/lifecycle/v2/ExperimentStatusResource.java]: Remove the `v2` as well keeping only the actual resources.

Comment (simonedavico) [inline application/src/main/java/cloud/benchflow/experimentsmanager/resources/lifecycle/v2/ExperimentStatusResource.java]: I don't know. We will probably need a `v3` for the new interactions from the orchestrator and the removal of the `deploy` phase.

Comment (VincenzoFerme) [inline application/src/main/java/cloud/benchflow/experimentsmanager/resources/lifecycle/v2/ExperimentStatusResource.java]: We can also do this change in a different branch, and when ready we merge everything back to the devel branch, removing/updating the current API.

**Comment by VincenzoFerme** [comment]
@simonedavico ping me when done

File: application/src/main/java/cloud/benchflow/experimentsmanager/resources/lifecycle/v2/ExperimentStatusResource.java (hunk 1 above)
Comment (VincenzoFerme) [inline application/src/main/java/cloud/benchflow/experimentsmanager/resources/lifecycle/v2/ExperimentStatusResource.java]: :+1:<a href='#crh-update-complete'></a>

**Comment by VincenzoFerme** [comment]
:+1:<a href='#crh-update-none'></a>

**Pull request merged**